doc/i2c_bus_lock_prevention/i2c_bus_lock_prevention_hld.md

```diff
@@ -0,0 +1,425 @@
+
+        """
+        pass
+
+    def reset_i2c_mux(self):
+        """
+        Reset I2C MUX to close all channels on the mux.
+
+        Returns:
+            bool: True if successful, otherwise the error result string.
+        """
+        pass
+
+    def representative_devices_get(self):
+        """
+        Return the representative i2c devices which will be probed in order to know
+        whether the i2c bus is locked.
+    
+        Returns:
+            list: A list of representative i2c devices
+        """
+        pass
+
+    def get_platform_service_list(self):
+       """
+       Return the services specific to run on a platform which will access i2c bus.
+
+       Returns:
+           list: A list of the platform specific services which will access i2c bus.
+       """
+       pass
+
+    def set_all_fan_full_speed(self):
+       """
+       Set all of the fan speed to full speed.
+
+       Returns:
+           bool: True if successful, otherwise the error result string.
+       """
+       pass
+
+    def set_i2c_faulty_device(self, bus, addr, faulty):
+        """
+        Set the faulty state for a device on the specified I2C bus and address.
+
+        Args:
+            bus (str): The I2C bus number.
+            addr (str): The I2C device address.
+            faulty (bool): True to mark the device as faulty, False otherwise.
+
+        Returns:
+            bool: True if the faulty state was set successfully, False otherwise.
+        """
+        pass
+```
+#### 7.6. New Table in STATE DB
+
+The i2c faulty devices detected by the i2c healthd will be written to the 'I2C_ISOLATION_LIST' table in STATE DB.
+
+Table Name: I2C_ISOLATION_LIST
+
+Key: I2C_ISOLATION_LIST|I2C Isolated Region ID|Isolated I2C Device Address
+
+Field: "device_name"
+
+Refer to [6.2. I2C Isolated Region](#62-i2c-isolated-region) for the consistent of the name in "I2C Isolated Region ID".
+"Isolated I2C Device Address" is the address of the isolated i2c device. The address is got from the platform API. The platform
+API may output the corresponding i2c device address under '/sys/bus/i2c/devices/'.
+
+Here is an example for the record in "I2C_ISOLATION_LIST" table in STATE DB.
+The table below contains one faulty device. The I2C Isolated Region ID of the device is "77-2-77-6"
+The isolated i2c device address in the region is "30-0050".
+The device name of the isolated i2c device is "Ethernet40". That means the transceiver inserted in
+the port Ethernet40 is being isolated.
+```
+admin@sonic:~$ redis-cli -n 6 hgetall "I2C_ISOLATION_LIST|77-2-72-6|30-0050"
+1) "device_name"
+2) "Ethernet40"
+```
+
+### 8. SAI API 
+
+SAI API is not used in this feature.
+
+### 9. Configuration and management 
+
+No configuration in this feature. 
+
+### 10. Warmboot and Fastboot Design Impact  
+
+The feature does not support warmboot and fastboot, and it has nothing to do with warmboot and fastboot.

```

Comment:
`i2chealthd` would get restarted during warm-boot and fast-reboot, right?
Does the I2C_ISOLATION_LIST need to be cleared, or will the isolation list be saved (i.e., do i2c lockups survive warm/fast boots)?